### ISSUE TITLE
Xml.decode doens't handle self-closing tags

### ISSUE THREAD

**Issue by mateusfccp** [eeue56/elm-xml#2]
In XML, when a tag has no children it can be written as self-closing. Actually, it's not only is it valid; sometimes the spec calls for it.<sup>[source](https://www.w3.org/TR/REC-xml/#sec-starttags)</sup> The syntax is as follows: `<tag></tag>` is equivalent to `<tag />`

Unfortunately, `Xml.decode` doesn't handle this possibility. For instance, if you try to decode
```
<tag>
    <autoclosing />
    <autoclosing />
</tag>
```

you get

```
StrNode "\n <autoclosing />\n <autoclosing />\n "
```

when it's supposed to get

```
Object (
    [ Tag "noautoclosing" (Dict.fromList []) (Object [])
    , Tag "noautoclosing" (Dict.fromList []) (Object []) ]
)
```

<hr />

You can get an reproducible example [here](https://ellie-app.com/Y4YFzczzMHa1/0).

**Comment by eeue56** [comment]
I think this would require adding a new type of tag. Do you want to make a PR? This should be a pretty simple change overall

**Comment by mateusfccp** [comment]
As soon as possible.

**Comment by mateusfccp** [comment]
@eeue56 I've managed to get a correct autoclose tag, but it only works when using a single autoclosing tag child. Actually, as the code is not comented, I found it a little confusing. What do you suggest?

I change in *Decode.elm* line 97

```
[] ->
    if String.startsWith "?" tagName then
        Ok ( DocType tagName props, firstClose + 1 )
    else
        "Failed to find close tag for "
            ++ tagName
            |> Err
```

to

```
[] ->
    if String.startsWith "?" tagName then
        Ok ( DocType tagName props, firstClose + 1 )
    else if (List.reverse words |> List.head |> Maybe.withDefault "") == "/" then
       Ok ( Tag tagName props (Object []), firstClose )
    else
        "Failed to find close tag for "
            ++ tagName
            |> Err
```

**Comment by eeue56** [comment]
@mateusfccp Open a PR and let's discuss there, it will be a little easier :)

**Comment by eeue56** [comment]
published as 2.1.1
